cwltool/sandboxjs.py

```diff
@@ -31,9 +49,11 @@ def new_js_proc():
 
     nodejs = None
     trynodes = ("nodejs", "node")
+    working_alias = None
     for n in trynodes:
         try:
             if subprocess.check_output([n, "--eval", "process.stdout.write('t')"]) != "t":
+                working_alias = n
```

Comment:
this branch means that node wasn't found (or found to be working for the trivial example)
I'd put your check for the correct version as the `else` to this `if` using `n` as your working_alias